test/test_dataloader2.py

```diff
@@ -104,6 +106,27 @@ def test_dataloader2_load_state_dict(self) -> None:
 
         restored_data_loader.shutdown()
 
+    def test_dataloader2_reset(self) -> None:
+
+        test_data_pipe = IterableWrapper(range(10))
+        reading_services = [None, TestReadingService(), MultiProcessingReadingService(num_workers=1)]
+
+        for reading_service in reading_services:
+            data_loader: DataLoader2 = DataLoader2(datapipe=test_data_pipe, reading_service=reading_service)
+
+            # Functional Test: Ensure multiple sequential reads of DL2 is possible
+            self.assertEqual(list(range(10)), list(data_loader))  # [0, ..., 10]`
+            self.assertEqual(list(range(10)), list(data_loader))  # [0, ..., 10]`
+            self.assertEqual(list(range(10)), list(data_loader))  # [0, ..., 10]`
+
+            # Functional Test: Ensure that the creation of a new iterator invalidates the old one
+            it1 = iter(data_loader)
+            self.assertEqual(0, next(it1))
+            it2 = iter(data_loader)
+            self.assertEqual(0, next(it2))
```

Comment:
Could you please run `next(it1)` more times before creating `it2`? I am worry about the result would always be `0`.
Same to `it2`, could you please run `list(it2)` to make sure the result is expected?